lib/cartopy/crs.py

```diff
@@ -53,7 +53,22 @@ def _safe_pj_transform(src_crs, tgt_crs, x, y, z=None, trap=True):
     if z is None:
         z = np.zeros_like(x)
 
-    return transformer.transform(x, y, z, errcheck=trap)
+    def _sanitize_scalars(a):
+        # future compatibility with numpy:
+        # pyproj implicitly converts size-1 arrays to scalars, which is
+        # deprecated in numpy 1.25
+        # see https://github.com/numpy/numpy/pull/10615
+        if isinstance(a, np.ndarray) and a.size == 1:
+            return a.item()
+        else:
+            return a
+
+    return transformer.transform(
```

Comment:
What about putting a context manager around a `filterwarnings` call here instead? The warning is sort of unfortunate and pyproj already has the workarounds for handling the single-item numpy arrays, so I guess I'd say we can just sweep the warning under the rug here because it is downstream and already taken care of.

```python
warnings.filterwarnings("ignore", message="Conversion of an array with ndim > 0")
```

Reply:
Let's try it